test/native/lib/TestEmitter.re

```diff
@@ -2,6 +2,34 @@ open Migrate_parsetree;
 open Ast_410;
 open Setup;
 
+let extract_tests = array_expr => {
+  open Ppxlib.Ast_pattern;
+  let fail = () =>
+    failwith("can only extract from array((result, expected))");
+  let payload = pexp_array(many(pexp_tuple(many(__))));
```

Comment:
This matching from ppxlib is really amazing a lot more easy than matching by hand on the AST, really recommend anyone interested to give it a look at the `ast_pattern.mli`